### ISSUE TITLE
Error callbacks

### ISSUE THREAD

**Issue by Talavera16** [etingof/pysnmp#230]
Hi!
I have a question about the use of pysnmp and I don´t know if a want to do is posible.
I want to throw raise of a known error code callback.
I´m using pysnmp like runway between a own protocol where I return a error code, if this matter, and this error codes are the same that in pysnmp:

SNMP_ERRORSTATUS_NOERROR0 | 
SNMP_ERRORSTATUS_TOOBIG1 |
SNMP_ERRORSTATUS_NOSUCHNAME2 |
SNMP_ERRORSTATUS_BADVALUE3 |
SNMP_ERRORSTATUS_READONLY4 |
SNMP_ERRORSTATUS_GENERR5
...

I´m triying to raise this errors with the codes of snmp in a handler where y throw GET and SET request.
but this always response General error, with the code 5 sawn in wireshark.
I thought that the error is in pysnmp/hlapi/asyncore/sync/cmdgen.py, but I don´t know how to use...

I would be very grateful if someone can help me

**Comment by etingof** [comment]
I am not sure I quite understand the work flow you are looking for. May be some code would help explaining your goal?

**Comment by Talavera16** [comment]
Sorry for my explanation. 
What I want, is to force to throw/raise a known SNMP error. 
My other application send to me, has occured a known SNMP error, but, this application doesn´t speak SNMP, and I use pysnmp to speak SNMP, and I want to throw a SNMP error, for example:

raise Exception(noSuchName)

and in pysnmp plot the error code were 2.
Thanks for your attemption.

**Comment by etingof** [comment]
Is it that you want to call pysnmp manager (client) from your application and in case of error response raise human-readable SNMP error?

**Comment by Talavera16** [comment]
Yes, I want to raise a SNMP error from my SNMO Agent. And for example, if I´m using a tool like MIB Browser, this show me the correct error.

**Comment by Talavera16** [comment]
Looking for, I saw in file "pysnmp/entity/rfc3413/cmdrsp.py" the next code:

class GetCommandResponder(CommandResponderBase):
    pduTypes = (rfc1905.GetRequestPDU.tagSet,)
    print ("[PX]:########### Pasa libreria por aqui 5")

    # rfc1905: 4.2.1
    def handleMgmtOperation(self, snmpEngine, stateReference,
                            contextName, PDU, acInfo):
        print ("[PX]:########### Pasa libreria por aqui 4")
        (acFun, acCtx) = acInfo
        # rfc1905: 4.2.1.1
        mgmtFun = self.snmpContext.getMibInstrum(contextName).readVars
        print("cosas que envia"+ str(mgmtFun) )
        self.sendVarBinds(snmpEngine, stateReference, 0, 0,
                          mgmtFun(v2c.apiPDU.getVarBinds(PDU), (acFun, acCtx)))
        self.releaseStateInformation(stateReference)
The two arguments 0, makes reference to errorStatus and errorIndex. My problem is, thath i Want to put in the function "sendVarBind" a kown errorStatus, but the previos class "GetCommandResponder" don´t allow to me do it.

How I can put in this errorStatus a value, without change the library...
Thanks for answers.

**Comment by lextm** [comment]
PySNMP allows you to use its predefined type `errorStatus` to translate numbers to strings if that's what you want.

https://github.com/lextudio/pysnmp/blob/v7.0.2/pysnmp/proto/rfc1905.py#L101

from pysnmp.proto.rfc1905 import errorStatus

name = errorStatus.namedValues.get(1)
print(name)  # Output: tooBig

With that you can throw an error like `raise Exception(name)`.

And if you want more than that, then you have to define your own type to translate numbers to strings where you have full control. Standards like RFC1905 won't help much in that case.
